internal/repository/location.go

```diff
@@ -0,0 +1,22 @@
+package repository
+
+import (
+	"go-rest-template/internal/domain"
+	"log"
+)
+
+type LocationRepository interface {
+	Create(location domain.Payload)
```

Comment:
Acá podés recibir el puntero, asi no se duplica en memoria.
```suggestion
	Create(p *domain.Payload)
```